report/renderer/performance-category-renderer.js

```diff
@@ -219,41 +219,42 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
     // Diagnostics
     const diagnosticAudits = category.auditRefs
         .filter(audit => this._isPerformanceInsight(audit))
-        .filter(audit => !ReportUtils.showAsPassed(audit.result))
-        .sort((a, b) => {
-          // Sort audits by impact, prioritizing those with a higher overallImpact first,
-          // then falling back to linearImpact, guidance level and score.
-          const {
-            overallImpact: aOverallImpact,
-            overallLinearImpact: aOverallLinearImpact,
-          } = this.overallImpact(a, metricAudits);
-          const {
-            overallImpact: bOverallImpact,
-            overallLinearImpact: bOverallLinearImpact,
-          } = this.overallImpact(b, metricAudits);
-
-          const aGuidanceLevel = a.result.guidanceLevel || 1;
-          const bGuidanceLevel = b.result.guidanceLevel || 1;
-
-          if (aOverallImpact !== bOverallImpact) return bOverallImpact - aOverallImpact;
-
-          if (
-            aOverallImpact === 0 && bOverallImpact === 0 &&
-            aOverallLinearImpact !== bOverallLinearImpact
-          ) {
-            return bOverallLinearImpact - aOverallLinearImpact;
-          }
+        .filter(audit => !ReportUtils.showAsPassed(audit.result));
+    /** @type {Array<{auditRef:LH.ReportResult.AuditRef, overallImpact: number, overallLinearImpact: number, guidanceLevel: number}>} */
+    const auditImpacts = [];
+    diagnosticAudits.forEach(audit => {
+      const {
+        overallImpact: overallImpact,
+        overallLinearImpact: overallLinearImpact,
+      } = this.overallImpact(audit, metricAudits);
+      // eslint-disable-next-line max-len
+      auditImpacts.push({auditRef: audit, overallImpact, overallLinearImpact, guidanceLevel: audit.result.guidanceLevel || 1});
```

Comment:
nit: can you do `const guidanceLevel = audit.result.guidanceLevel || 1` on a separate line? Would also reduce the line length :)